### PR TITLE
Add unit test file for passwd in envars.

### PR DESCRIPTION
**Pull request by y-cote**
1- The test detects environment variable populated with specific values, but will only log any hits and NOT make the test fail at the moment. This could be changed in the future.
2- The white list is used to deal with false positives OR yaml configs we prefer not to change and continue using environment variables.
3- The test is extensible by adding new routines starting with "checkX()".
 
```release-note
NONE
```

### CODE REVIEW

**Comment by mjudeikis** [comment]
rebase is needed.
What is the background for this PR?

**Comment by thekad** [comment]
I am guessing it is related to AZURE-324

File: pkg/sync/v9/envpass_test.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+		}
+	}
+	return false
+}
+
+// We are looking for a structure like this:
+// env:
+// - name: SomeVarName
+//   value: '*** GENERATED ***'
+func checkValueGenerated(path string, o unstructured.Unstructured, t *testing.T) bool {
```
Comment (asalkeld) [inline pkg/sync/v9/envpass_test.go]: t is not used (maybe some debugging?)

File: pkg/sync/v9/envpass_test.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+		if r := checkValueGenerated(path, u, t); r == true {
+			t.Logf("%s: contains '*** GENERATED ***' in env[*].value which probably means that secrets are passed in environment variables. This is not allowed for security reasons.", path)
+		}
+
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(paths) > 0 {
```
Comment (asalkeld) [inline pkg/sync/v9/envpass_test.go]: paths is never set

File: pkg/sync/v9/envpass_test.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+		u, err := unmarshal(b1)
+		if err != nil {
+			return err
+		}
+
+		if r := checkSecretKeyRef(path, u); r == true {
+			t.Logf("%s: contains 'password' in env[*].valueFrom.secretKeyRef.key which probably means that secrets are passed in environment variables. This is not allowed for security reasons.", path)
+		}
+		if r := checkValueGenerated(path, u, t); r == true {
+			t.Logf("%s: contains '*** GENERATED ***' in env[*].value which probably means that secrets are passed in environment variables. This is not allowed for security reasons.", path)
```
Comment (asalkeld) [inline pkg/sync/v9/envpass_test.go]: these never result in the test failing
were you going to do something like this below?
```
paths = append(paths, path)
```

File: pkg/sync/v9/envpass_test.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+		}
+
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(paths) > 0 {
+		t.Errorf("invalid files (rerun with REGENERATE environment variable set):\n  %s", strings.Join(paths, "\n  "))
```
Comment (asalkeld) [inline pkg/sync/v9/envpass_test.go]: this error message is misleading. i don't think that running REGENERATE will fix the password being passed in.

**Comment by asalkeld** [comment]
I have rebased and fixed (kept @y-cote 's commit). Moved to https://github.com/openshift/openshift-azure/pull/1954

**Pull request closed without merging**